nginx.tmpl

```diff
@@ -549,7 +571,9 @@ server {
     {{- if $server_tokens }}
     server_tokens {{ $server_tokens }};
     {{- end }}
+    {{- if $http2 }}
```

Comment:
http2 disabling will not work because it's enabled in the fallback block.
Nginx creates listen processes for for each ip:port combination and then sums the capabilities of every host. So the `http2 on` from the fallback does globally turn it on.

So the label on the **proxied** container will never work. If 1 vhost is http2 enabled, it will be for all. I think that can be removed.
The environment on the **proxy** container could work. But then [line #446](https://github.com/nginx-proxy/nginx-proxy/blob/12653debb5562c227a5f1ac5703bb942c6167173/nginx.tmpl#L446) also needs to be behind the if statement.

Reply:
Thanks for those information, I'll keep the global variable with the needed modification and remove the proxied container label.

Reply:
I was thinking some more and maybe its better to not have the option to disable http2 at all.
All currently supported QUIC browsers seem to connect over http1/http2 first.
When the alt-svc header is provided, the browser upgrades to QUIC and caches that for the time specified in alt-svc.
After that cache period a tcp (http2) request is done to negotiate this again.

And for the next foreseeable future you probably want http2 as fallback and for now adds unnecessary overhead.

80 > 443 (http2) > alt-svc (http2) > QUIC (http3)

Reply:
The HTTP/2 disabling feature is a separate request that people made specifically (see https://github.com/nginx-proxy/nginx-proxy/pull/1644#issuecomment-851297376), it was added to this PR because it kind of made sense but it does not imply in any way that you have to or even should disable HTTP/2 when using HTTP/3.

Reply:
Additional request for HTTP/2 disabling feature : https://github.com/nginx-proxy/nginx-proxy/issues/1055

Reply:
After testing per virtual host HTTP/2 disabling, it does appear to work as intended. Requests made with `curl --http2`, www.foo.buz has HTTP/2 enabled (default), www.foo.bar has HTTP/2 disabled via the label:
```console
nginx-proxy          | nginx.1     | www.foo.buz 172.22.0.1 - - [22/Jul/2023:08:35:17 +0000] "GET / HTTP/2.0" 200 615 "-" "curl/8.2.0-DEV" "172.22.0.2:80"
nginx-proxy          | nginx.1     | www.foo.bar 172.22.0.1 - - [22/Jul/2023:08:35:22 +0000] "GET / HTTP/1.1" 200 615 "-" "curl/8.2.0-DEV" "172.22.0.3:80"
```